src/app/components/layout/LanguagePickerDialog.js

```diff
@@ -0,0 +1,121 @@
+
+const messages = defineMessages({
+  optionLabel: {
+    id: 'languagePickerDialog.optionLabel',
+    defaultMessage: '{languageName} ({languageCode})',
+  },
+});
+
+const LanguagePickerDialog = ({
+  intl,
+  isSaving,
+  onDismiss,
+  onSubmit,
+  open,
+  team,
+}) => {
+  const [value, setValue] = React.useState(null);
+  const languages = safelyParseJSON(team.get_languages) || [];
+
+  const options = (languages ? languages.concat('disabled') : [])
+    .concat(Object.keys(LanguageRegistry)
+      .filter(code => !languages.includes(code)));
+
+  // intl.formatMessage needed here because Autocomplete
+  // performs toLowerCase on strings for comparison
+  const getOptionLabel = (code) => {
+    if (code === 'disabled') return '──────────';
+
+    return intl.formatMessage(messages.optionLabel, {
+      languageName: (
+        LanguageRegistry[code] ?
+          `${LanguageRegistry[code].name} / ${languageLabel(code)}` :
+          `${languageLabel(code)}`
+      ),
+      languageCode: code,
+    });
+  };
+
+  const handleChange = (e, val) => {
+    setValue(val);
+  };
+
+  const handleSubmit = () => {
+    onSubmit({ languageCode: value, languageName: languageLabel(value) });
+  };
+
+  return (
+    <Dialog
+      open={open}
+      maxWidth="xs"
+      fullWidth
+    >
+      <DialogTitle>
+        <FormattedMessage id="languagePickerDialog.title" defaultMessage="Choose a language" />
+      </DialogTitle>
+      <DialogContent>
+        <Autocomplete
+          id="autocomplete-add-language"
+          name="autocomplete-add-language"
+          options={options}
+          openOnFocus
+          getOptionLabel={getOptionLabel}
+          getOptionDisabled={option => option === 'disabled'}
+          value={value}
+          renderInput={
+            params => (<TextField
+              label={
+                <FormattedMessage
+                  id="languagePickerDialog.selectLanguage"
+                  defaultMessage="Select a language"
+                />
+              }
+              {...params}
+            />)
+          }
+          onChange={handleChange}
+          fullWidth
+        />
+      </DialogContent>
+      <DialogActions>
+        <Button className="add-language-action__cancel" onClick={onDismiss}>
+          <FormattedGlobalMessage messageKey="cancel" />
+        </Button>
+        <Button
+          className="add-language-action__submit"
+          color="primary"
+          endIcon={isSaving ? <CircularProgress color="inherit" size="1em" /> : null}
+          disabled={!value || isSaving}
+          onClick={handleSubmit}
+          variant="contained"
+        >
+          <FormattedGlobalMessage messageKey="submit" />
+        </Button>
+      </DialogActions>
+    </Dialog>
+  );
+};
+
+LanguagePickerDialog.propTypes = {
+  team: PropTypes.shape({
+    id: PropTypes.string.isRequired,
+    get_languages: PropTypes.string.isRequired,
+  }).isRequired,
+  intl: intlShape.isRequired,
+};
```

Comment:
What about the other props?
-   isSaving
-   onDismiss
-   onSubmit
-   open

Reply:
Missed these, thanks! Fixed